### ISSUE TITLE
Striping docker.io off docker tags

### ISSUE THREAD

**Issue by kellyjgibson** [aztechian/bridgr#58]
Bridgr is stripping the `docker.io/` off image tags despite it being explicitly stated in my manifest. For my particular application, I need the leading repository preserved.

**Comment by aztechian** [comment]
Can you provided a minimal debug output?
Something like `bridgr -v <config> docker`
I need to know at what "part" of the process the tagging isn't happening. It's possible it may be whats coming out of the `docker save` (equivalent) command, so we may also need to check the manifests in the .tar thats saved.

**Comment by kellyjgibson** [comment]
Workaround:
```bash
PRE_TAG=`docker images --format "{{.Repository}}:{{.Tag}}" | grep -v '<none>' | grep -v '.io'`

for IMAGE in ${PRE_TAG[@]}; do
  docker tag ${IMAGE} docker.io/${IMAGE}
done```

**Comment by kellyjgibson** [comment]
Debug Output:

>> ./bridgr -v -c bridge.ign.yml docker
2019/10/17 09:45:34 Final Files configuration {Items:[]}
2019/10/17 09:45:34 Unknown configuration section for Python: %!s(<nil>)
2019/10/17 09:45:34 Final Python configuration {Items:[] Image:docker.io/library/python:2}
2019/10/17 09:45:34 Unknown configuration section for Ruby: %!s(<nil>)
2019/10/17 09:45:34 Final Ruby configuration {Items:[] Sources:[https://rubygems.org] Image:docker.io/library/ruby:2-alpine}
2019/10/17 09:45:34 Final Git configuration {Items:[]}
2019/10/17 09:45:34 Created ..../platform/packages/bridgr.repo for writing repo template
2019/10/17 09:45:34 Created ..../platform/packages/requirements.txt for writing repo template
2019/10/17 09:45:34 Created ..../platform/packages/Gemfile for writing Gemfile template
2019/10/17 09:45:34 Running workers for subcommands: [docker]
2019/10/17 09:45:34 Processing Docker...
2019/10/17 09:45:34 Called Docker.Setup()
2019/10/17 09:45:34 pulling image docker.io/openshift/origin-control-plane:v3.11
2019/10/17 09:45:34 Got image domain of https://docker.io
2019/10/17 09:45:34 Parsed URL: https://docker.io
2019/10/17 09:45:34 Looking for env var: BRIDGR_DOCKER_IO_USER
2019/10/17 09:45:34 Env Var BRIDGR_DOCKER_IO_USER was not found :(
{"status":"Pulling from openshift/origin-control-plane","id":"v3.11"}
{"status":"Digest: sha256:01254cfe64294603fa31f2610a4f601e18b4f32b1e6a933d35ba583f7fb5ea8c"}
{"status":"Status: Image is up to date for openshift/origin-control-plane:v3.11"}
2019/10/17 09:45:35 pulling image docker.io/openshift/origin-pod:v3.11
2019/10/17 09:45:35 Got image domain of https://docker.io
2019/10/17 09:45:35 Parsed URL: https://docker.io
2019/10/17 09:45:35 Looking for env var: BRIDGR_DOCKER_IO_USER
2019/10/17 09:45:35 Env Var BRIDGR_DOCKER_IO_USER was not found :(
{"status":"Pulling from openshift/origin-pod","id":"v3.11"}
{"status":"Digest: sha256:e1318e028e1d6f7e6b70b0d5039f0e79177dd8d07fa9f32d3e931ea72d0a8c81"}
{"status":"Status: Image is up to date for openshift/origin-pod:v3.11"}
2019/10/17 09:46:04 saved Docker image docker.io/openshift/origin-control-plane:v3.11 to /...../platform/packages/docker/openshift_origin-control-plane.tar

**Comment by aztechian** [comment]
> 2019/10/17 09:45:34 pulling image docker.io/openshift/origin-control-plane:v3.11
{"status":"Pulling from openshift/origin-control-plane","id":"v3.11"}

This is concerning ... the first line is from bridgr (it uses a fully canonicalized data structure internally to reference images) the second is output from the docker daemon: it looks like it throws out the domain if its the default registry source. 

For your workaround above: it seems like this is tagging anything in the local daemon that isn't fully canonical to have `docker.io` put in? This is assuming that the local daemon already has all the images?
If this works for whats exported (to be properly referenced in the tar), bridgr should be able to re-tag all images to be fully canonical before saving them from the local daemon.

**Comment by kellyjgibson** [comment]
Correct, I'm modifying them at the local destination daemon to have the "correct" tag I need.  This was the shortest fix for my use case (K8S starts these docker images).
